### DISCUSSION TITLE
Configure Email Notifications for Discussions

### DISCUSSION THREAD

**Discussion by cometz95** [community/community#62311] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello,

I would like to configure my email notifications for discussions to be slightly more customized, rather than receiving an email every time there is a new post (even if it's a comment). Can I configure it so I only receive notifications for certain categories within my Discussion board? Notifications only when there's a new post (rather than comments on a new post)? I would like to allow my team members to customize their notification settings in this way too, not just me as the owner.

If there's no built in way to do this, perhaps a workaround is to setup some kind of email bot and use info from the email header? 
(https://docs.github.com/en/account-and-profile/managing-subscriptions-and-notifications-on-github/setting-up-notifications/configuring-notifications#filtering-email-notifications)
Perhaps I could use this to route all email notifications to one email address and then route specific emails to members of my team.

If anyone has done this kind of thing before or could point me in the right direction, that would be a big help.

Thanks in advance!

**Comment by sandroamiridze** [comment, marked as answer]
Hello,

To achieve the level of customization you're looking for in email notifications for discussions, you can follow the workaround you mentioned with the email bot. Although there might not be a built-in way to configure notifications to the extent you desire within the Discussion board on GitHub, you can use an email bot to filter and route specific emails to you and your team members based on categories or other criteria.

Here's how you can set up the email bot and route emails effectively:

1. Set up an Email Bot:
   - You can use third-party email services like Zapier, IFTTT, or even create a custom email bot using programming languages like Python.
   - Configure the bot to receive emails from the GitHub notifications email address.

2. Customize Email Filtering:
   - With the email bot in place, you can customize the filtering rules to process incoming emails based on the subject, content, or sender address. This allows you to sort emails into different categories.

3. Route Specific Emails to Team Members:
   - Once emails are categorized, you can set up rules in the email bot to forward specific emails to different team members based on their preferences.
   - If you have distinct categories in your Discussion board, you can forward emails from each category to the relevant team members.

4. Customize Team Members' Settings:
   - Each team member can then customize their own email bot settings to receive notifications for the categories they are interested in.

By implementing this email bot setup, you and your team members can receive more customized notifications, only getting notified about the specific topics or categories you choose. It's essential to choose a reliable and secure email service or bot to handle sensitive data.

Remember that this workaround involves handling email forwarding and filtering externally from the GitHub platform, so ensure you follow security best practices and protect sensitive information. Always test the setup thoroughly to ensure it functions as expected and doesn't lead to any unintended consequences.

If you're not familiar with creating email bots, you can refer to the documentation and support resources provided by the chosen email service or tool for guidance on setting up filters and routing emails.

I hope this points you in the right direction, and good luck with customizing your team's notification settings!

**Reply by cometz95** [reply]
Great thank you so much!

**Reply by cometz95** [reply]
Hi again Sandro, I was wondering if you have any further info on how to forward emails only from a specific category. When I receive an email from a discussion post, the category it was posted in does not appear to be contained in the email anywhere, so I don't know how to tell the email bot which category it was posted in (thus I wouldn't know how to route an email from a specific category). Thanks

**Reply by sandroamiridze** [reply]
Hello,

If the category information is not readily available in the email content, you might need to explore alternative methods to identify the category of a discussion post. One approach is to parse the email subject line, as it often contains relevant details.

Here are some steps you can take to extract the category information from the email subject:

1. Analyze the email subject line structure: Look for patterns in the subject line that might indicate the category. For example, if the subject line includes a specific keyword or tag related to the category.

2. Use regular expressions: Create regular expressions to match the category-related information in the subject line. Regular expressions allow you to search for specific patterns in text strings.

3. Extract the category: Once you have identified the category-related text using regular expressions, you can extract and store that information to determine the category of the discussion post.

4. Route emails based on the extracted category: Use the extracted category information to set up rules in your email bot for forwarding emails to the relevant team members or handling them according to their category.

Keep in mind that the effectiveness of this method may depend on the consistency of the subject line format used in the email notifications from GitHub. Be sure to thoroughly test the regular expressions and email filtering rules to ensure accurate categorization.

If you need assistance with writing regular expressions or setting up the email bot, you can consult the documentation and support resources of the email service or tool you are using. Additionally, there are online communities and forums where you can seek advice and help from experienced users.

I hope this helps you in achieving the desired email categorization. If you have any further questions or need more guidance, feel free to ask!

If the category information is not included in the email content, you may have to look for alternative ways to categorize and forward those emails. One approach could be using a keyword-based system. 

For example, you could include a specific keyword in the subject line of discussion post emails, like "[Discussion]" or something similar. Then, you can set up an email filter or rule to forward emails containing that keyword to a designated email address or folder.

If you are using GitHub's email integration, you can check if it allows filtering based on subject keywords. Additionally, some email services provide APIs that can be used to programmatically forward emails based on certain criteria.

Keep in mind that the exact process might differ based on your email provider and the tools you are using. Make sure to explore the options available in GitHub's email integration or consult their documentation for more specific guidance.

**Reply by cometz95** [reply]
Hey, the name of the category is not in the subject line by default and I don't think I have the ability to modify the subject line or the contents of the email. I don't think keyword filtering is available by default, and I don't think Github as an API to setup email notifications.
